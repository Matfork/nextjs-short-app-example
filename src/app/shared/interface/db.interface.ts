interface DbPossibleValues {
  id: number;
  value: string;
  description: string;
  isDefault: boolean;
}

export interface DbMenu {
  value: number;
  label: string;
}

export interface DbDescription {
  id: number;
  keyName: string;
  description: string;
  type: number;
  isPersonalData: boolean;
  possibleValues: DbPossibleValues[];
}

export interface DbRequest {
  id: number;
  date: string;
  reason: string;
  status: string;
}
