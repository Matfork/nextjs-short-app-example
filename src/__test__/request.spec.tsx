/* eslint-env jest */
import React from 'react';
import RequestListRequestComponent from '../../src/app/modules/components/request/RequestList.request.component';
import { mount } from 'enzyme';
import { DatabaseStorageService } from '../../src/app/shared/services/DataStorage.service';
import { DbRequest } from '../app/shared/interface/db.interface';

describe('Request Page', () => {
  describe('All Request', () => {
    it('should display pending request', async function() {
      await DatabaseStorageService.initialize();
      const wrap = mount(<RequestListRequestComponent />);
      expect(wrap.find('tbody.table-body').html()).toMatchSnapshot();
    });
  });

  describe('Filtered denied Request', () => {
    it('should display denied request', async function() {
      await DatabaseStorageService.initialize();

      const pr: DbRequest[] = DatabaseStorageService.getTableData('requests');
      const filered = pr.filter((el: any) => el.status === 'denied');

      // doenst work when Component to spy is a functional component instead of a class component
      //   const spy = jest.spyOn(
      //     RequestListRequestComponent.prototype,
      //     'handleChange'
      //   );
      const wrap = mount(<RequestListRequestComponent />);

      wrap
        .find('div.MuiTabs-flexContainer button')
        .at(3)
        .simulate('click');

      //   expect(spy).toHaveBeenCalledTimes(1);

      expect(wrap.find('tbody.table-body tr')).toHaveLength(filered.length);
      expect(wrap.find('tbody.table-body').html()).toMatchSnapshot();
    });
  });
});
