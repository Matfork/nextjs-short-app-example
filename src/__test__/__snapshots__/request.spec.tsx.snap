// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Page All Request should display pending request 1`] = `"<tbody class=\\"MuiTableBody-root table-body\\"><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/09/21</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Data science algorithms</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(0, 0, 0);\\">pending</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/08/13</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Stakeholder dashboards</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(85, 108, 214);\\">approved</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/07/03</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Email Blast</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: red;\\">denied</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/02/09</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Investigation</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(85, 108, 214);\\">approved</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/09/21</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Data science algorithms</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(0, 0, 0);\\">pending</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/08/13</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Stakeholder dashboards</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(85, 108, 214);\\">approved</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/07/03</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Email Blast</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: red;\\">denied</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/02/09</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Investigation</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(85, 108, 214);\\">approved</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/09/21</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Data science algorithms</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(0, 0, 0);\\">pending</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/08/13</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Stakeholder dashboards</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(85, 108, 214);\\">approved</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/07/03</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Email Blast</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: red;\\">denied</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/02/09</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Investigation</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: rgb(85, 108, 214);\\">approved</span></td></tr></tbody>"`;

exports[`Request Page Filtered denied Request should display denied request 1`] = `"<tbody class=\\"MuiTableBody-root table-body\\"><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/07/03</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Email Blast</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: red;\\">denied</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/07/03</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Email Blast</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: red;\\">denied</span></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body\\" scope=\\"row\\">2017/07/03</th><td class=\\"MuiTableCell-root MuiTableCell-body\\">Email Blast</td><td class=\\"MuiTableCell-root MuiTableCell-body\\"><span class=\\"status\\" style=\\"background-color: red;\\">denied</span></td></tr></tbody>"`;
